engine/engine.go

```diff
@@ -135,12 +135,22 @@ func (e *Docker) Destroy(ctx context.Context, specv runtime.Spec) error {
 
 	// stop all containers
 	for _, step := range append(spec.Steps, spec.Internal...) {
-		e.client.ContainerKill(ctx, step.ID, "9")
+		if err := e.client.ContainerKill(ctx, step.ID, "9"); err != nil {
+			logger.FromContext(ctx).
+				WithError(err).
+				WithField("container", step.ID).
+				Debugln("cannot kill container")
+		}
 	}
 
 	// cleanup all containers
 	for _, step := range append(spec.Steps, spec.Internal...) {
-		e.client.ContainerRemove(ctx, step.ID, removeOpts)
+		if err := e.client.ContainerRemove(ctx, step.ID, removeOpts); err != nil {
```

Comment:
```suggestion
		if err := e.client.ContainerRemove(ctx, step.ID, removeOpts); err != nil && !client.IsErrNotFound(err) {
```

Because some containers might not exist (skipped steps).